pkgs/os-specific/darwin/mousecape/default.nix

```diff
@@ -0,0 +1,37 @@
+{ lib
+, stdenvNoCC
+, fetchurl
+, unzip
+}:
+
+stdenvNoCC.mkDerivation rec {
+  pname = "mousecape";
+  version = "1813";
+
+  src = fetchurl {
+    url = "https://github.com/alexzielenski/Mousecape/releases/download/${version}/Mousecape_${version}.zip";
+    hash = "sha256-lp7HFGr1J+iQCUWVDplF8rFcTrGf+DX4baYzLsUi/9I=";
+  };
+
+  sourceRoot = ".";
+
+  nativeBuildInputs = [ unzip ];
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/Applications
+    mv Mousecape.app $out/Applications
+
+    runHook postInstall
+  '';
+
+  meta = with lib; {
```

Comment:
Missing `meta.mainProgram`

Reply:
Nice to have:
1. Remove `with lib;`
2. Sort meta attributes alphabetically

Reply:
Hey! First of all, I want to thank you a lot for explaining things in a simple manner and providing me with resources. I'm still pretty new to Nix, so this is super helpful.

I have two questions, though. You say that I'm missing `meta.mainProgram`, which is true. But, when I looked at the other programs in the `darwin` folder, none of them have it. 🤔

Also, I'm curious, why remove `with lib;`? Isn't it better that way instead of having `with lib;` on every single property inside meta?

Reply:
Hello there,

Maintainers are here to help, so thank you for your kind words, much appreciated.

Replying to your questions now:

1. No, your assertion is not true. Counter example: https://github.com/DontEatOreo/nixpkgs/blob/91f212d4699be23336ec13f738227b6dbfa6f802/pkgs/os-specific/darwin/apparency/default.nix#L35
2. It's a good practice. Technically, using `with lib;` is fine. There are [initiative](https://github.com/NixOS/nixpkgs/issues/208242) cleaning toplevel `with lib;`, but I somehow enforce it when it is applied to sets when doing reviews.

Reply:
I'm looking at https://github.com/DontEatOreo/nixpkgs/blob/91f212d4699be23336ec13f738227b6dbfa6f802/pkgs/os-specific/darwin/apparency/default.nix#L35 and I see that it's using `appy` (even though the name of the program is `Apparency.app`), and I see it's derived from here.

![image](https://github.com/NixOS/nixpkgs/assets/57304299/b320d358-8ad9-4759-95dd-00ae6ea79941)

There is no way to use Mousecape through the terminal, so I'm not too sure what the `mainProgram` should be here. Maybe `Mousecape.app`?

Reply:
I don't have Darwin machine to test, maybe you can test and report back?

Reply:
It builds and runs properly regardless of missing `mainProgram` or using `mainProgram = "mousecape";`.

I'm going to keep `mainProgram = "mousecape";` as you request having `meta.mainProgram`

Reply:
When you do: `nix run .#mousecape`, does it run ? If not, please remove `meta.mainProgram` then.

Reply:
It didn't ran so, no `meta.mainProgram` then.

Reply:
Fair enough, thanks! Today I learned :)